src/azul/plugins/metadata/hca/transform.py

```diff
@@ -350,17 +354,22 @@ def for_id(cls, submitter_id: str) -> Optional['Submitter']:
         except KeyError:
             return None
 
+    @classmethod
+    def for_title(cls, title: str) -> Optional['Submitter']:
+        try:
+            submitter_id = cls.id_by_title[title]
+        except KeyError:
+            return None
+        else:
+            return cls.for_id(submitter_id)
```

Comment:
If you track the enum instance, ot just the ID, in the map, you wouldn't need this 2nd lookup.